hotel_housekeeping/models/hotel_housekeeping.py

```diff
@@ -0,0 +1,234 @@
+
+    current_date = fields.Date("Today's Date", required=True,
+                               index=True,
+                               states=DONE_STATE,
+                               default=(lambda *a:
+                                        time.strftime
+                                        (DEFAULT_SERVER_DATE_FORMAT)))
+    clean_type = fields.Selection([('daily', 'Daily'),
+                                   ('checkin', 'Check-In'),
+                                   ('checkout', 'Check-Out')],
+                                  'Clean Type', required=True,
+                                  states=DONE_STATE,)
+    room_no = fields.Many2one('hotel.room', 'Room No', required=True,
+                              states=DONE_STATE,
+                              index=True)
+    activity_lines = fields.One2many('hotel.housekeeping.activities',
+                                     'a_list', 'Activities',
+                                     states=DONE_STATE,
+                                     help='Detail of housekeeping activities',)
+    inspector = fields.Many2one('res.users', 'Inspector', required=True,
+                                index=True,
+                                states=DONE_STATE)
+    inspect_date_time = fields.Datetime('Inspect Date Time', required=True,
+                                        states=DONE_STATE)
+    quality = fields.Selection([('excellent', 'Excellent'), ('good', 'Good'),
+                                ('average', 'Average'), ('bad', 'Bad'),
+                                ('ok', 'Ok')], 'Quality',
+                               states=DONE_STATE,
+                               help="Inspector inspect the room and mark \
+                                as Excellent, Average, Bad, Good or Ok. ")
+    state = fields.Selection([('inspect', 'Inspect'), ('dirty', 'Dirty'),
+                              ('clean', 'Clean'),
+                              ('done', 'Done'),
+                              ('cancel', 'Cancelled')], 'State',
+                             states=DONE_STATE,
+                             index=True, required=True, readonly=True,
+                             default=lambda *a: 'inspect')
+
+    @api.multi
+    def action_set_to_dirty(self):
+        """
+        This method is used to change the state
+        to dirty of the hotel housekeeping
+        ---------------------------------------
+        @param self: object pointer
+        """
+        self.state = 'dirty'
+        for line in self:
+            line.quality = False
+            for activity_line in line.activity_lines:
+                activity_line.clean = False,
+                activity_line.dirty = True
+
+    @api.multi
+    def room_cancel(self):
+        """
+        This method is used to change the state
+        to cancel of the hotel housekeeping
+        ---------------------------------------
+        @param self: object pointer
+        """
+        self.state = 'cancel'
```

Comment:
you could save an extra `write()` operation if you use it explicitly, like 
```python
self.write({'state': 'cancel', 'quality': False})
```
this will work on multrecords as well